src/components/Profile/EditProfileModal.tsx

```diff
@@ -73,9 +101,15 @@ export default function EditProfileModal() {
               />
             </div>
             <button
+              disabled={mutation.isLoading}
               className="btn mt-6 bg-primary px-3 text-white"
-              onClick={() => onUpdateClick()}
+              onClick={(e) => onUpdateClick(e)}
```

Comment:
This button is inside of a form so whenever it is clicked by default the form will be submitted.
Currently it looks like you're trying to prevent default with the onUpdateClick function, however, the event in this case is the button click, not the submit form event. 
To prevent the form submission (and page reload) on this button press this function should be moved to the form's `onSubmit` handler. Then the event will be the form submission event and prevent default will work as intended.